logic/strategy/calculated_crib.go

```diff
@@ -1,32 +1,37 @@
 package strategy
 
 import (
+	"errors"
+
 	"github.com/joshprzybyszewski/cribbage/logic/scorer"
 	"github.com/joshprzybyszewski/cribbage/model"
 )
 
 // GiveCribHighestPotential gives the crib the highest potential pointed crib
-func GiveCribHighestPotential(_ int, hand []model.Card) []model.Card {
+func GiveCribHighestPotential(_ int, hand []model.Card) ([]model.Card, error) {
 	isBetter := func(old, new float64) bool { return new > old }
 	return getBestPotentialCrib(hand, isBetter)
 }
 
 // GiveCribLowestPotential gives the crib the lowest potential pointed hand
-func GiveCribLowestPotential(_ int, hand []model.Card) []model.Card {
+func GiveCribLowestPotential(_ int, hand []model.Card) ([]model.Card, error) {
 	isBetter := func(old, new float64) bool { return new < old }
 	return getBestPotentialCrib(hand, isBetter)
 }
 
-func getBestPotentialCrib(hand []model.Card, isBetter func(old, new float64) bool) []model.Card {
+func getBestPotentialCrib(hand []model.Card, isBetter func(old, new float64) bool) ([]model.Card, error) {
 	if len(hand) > 6 || len(hand) <= 4 {
-		return nil
+		return nil, errors.New(`invalid input`)
```

Comment:
Could we have this be a more descriptive error message? Maybe `hand size must be from 4 to 6` -- it makes tracking errors down easier if they're specific to their case